routecontroller/resourcebuilders/http_proxy_builder.go

```diff
@@ -0,0 +1,144 @@
+package resourcebuilders
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	networkingv1alpha1 "code.cloudfoundry.org/cf-k8s-networking/routecontroller/apis/networking/v1alpha1"
+	hpv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type HTTPProxyBuilder struct {
+	TLSSecretName string
+}
+
+func HTTPProxyName(fqdn string) string {
+	sum := sha256.Sum256([]byte(fqdn))
+	return fmt.Sprintf("hp-%x", sum)
+}
+
+func (b *HTTPProxyBuilder) BuildMutateFunction(actual, desired *hpv1.HTTPProxy) controllerutil.MutateFn {
+	return func() error {
+		actual.ObjectMeta.Labels = desired.ObjectMeta.Labels
+		actual.ObjectMeta.Annotations = desired.ObjectMeta.Annotations
+		actual.ObjectMeta.OwnerReferences = desired.ObjectMeta.OwnerReferences
+		actual.Spec = desired.Spec
+		return nil
+	}
+}
+
+func (b *HTTPProxyBuilder) Build(routes *networkingv1alpha1.RouteList) ([]hpv1.HTTPProxy, error) {
+	resources := []hpv1.HTTPProxy{}
+
+	routesForFQDN := groupByFQDN(routes)
+	sortedFQDNs := sortFQDNs(routesForFQDN)
+
+	for _, fqdn := range sortedFQDNs {
+		httpProxy, err := b.fqdnToHTTPProxy(fqdn, routesForFQDN[fqdn])
+		if err != nil {
+			return []hpv1.HTTPProxy{}, err
+		}
+
+		resources = append(resources, httpProxy)
+	}
+
+	return resources, nil
+}
+
+func (b *HTTPProxyBuilder) fqdnToHTTPProxy(fqdn string, routes []networkingv1alpha1.Route) (hpv1.HTTPProxy, error) {
+	name := HTTPProxyName(fqdn)
+	hp := hpv1.HTTPProxy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: routes[0].ObjectMeta.Namespace,
+			Labels:    map[string]string{},
+			Annotations: map[string]string{
+				"cloudfoundry.org/fqdn": fqdn,
+			},
+			OwnerReferences: []metav1.OwnerReference{},
+		},
+		Spec: hpv1.HTTPProxySpec{
+			VirtualHost: &hpv1.VirtualHost{
+				Fqdn: fqdn,
+				TLS: &hpv1.TLS{
```

Comment:
AFAIK, this is not going to work with wildcard hosts, we need to use [TLS Certificate Delegation](https://projectcontour.io/docs/v1.9.0/httpproxy/#tls-certificate-delegation)